cow_amm/cow_amm_lp_and_tvl_4047078.sql

```diff
@@ -0,0 +1,174 @@
+-- Computes the TVL, lp token total supply and lp token price of a CoW AMM pool over time
+-- Parameters
+--  {{token_a}} - either token of the desired uni pool
+--  {{token_b}} - other token of the desired uni pool
+--  {{start}} - date as of which the analysis should run
+
+-- Given that we might not have records every day in the source data (e.g. not every day the lp supply may change), 
+-- but still want to visualize development on a per day basis,  we create an auxiliary table with one record per 
+-- day between `start` and `now`
+with date_range as (
+    select t.day
+    from
+        unnest(sequence(
+            date(timestamp '{{start}}'),
+            date(now())
+        )) t (day) --noqa: AL01
+),
+
+-- Finds the CoW AMM pool address given tokens specified in query parameters (regardless of order) 
+cow_amm_pool as (
+    select distinct
+        cowamm_address as address,
+        blockchain,
+        token_1_address,
+        token_2_address
+    from (
+        select
+            t.blockchain,
+            cowamm_address,
+            token_1_reserve as token_1_balance,
+            token_2_reserve as token_2_balance,
+            t.token_1_address,
+            t.token_2_address,
+            count(coalesce(token_1_reserve, token_2_reserve)) over (partition by cow_amm_nb order by "time" desc) as "temp"
+        from query_3959058 as t
+        left join
+            (select
+                *,
+                count(address) over (order by address, token_1_address, token_2_address) as cow_amm_nb
+            from query_3959044) as p
+            on
+                t.cowamm_address = p.address and t.blockchain = p.blockchain
+                and t.token_1_address = p.token_1_address and t.token_2_address = p.token_2_address
+
+        where
+            ((t.token_1_address = {{token_a}} and t.token_2_address = {{token_b}}) or (t.token_2_address = {{token_a}} and t.token_1_address = {{token_b}}))
+    ) where temp = 1
+    and token_1_balance > 0 and token_2_balance > 0
+),
```

Comment:
Not sure it it makes sense to keep this complex logic to avoid empty test pools. 
The previous query shows the test pools, but it should not affect the result as they are empty.

Reply:
What is this query doing (the comment was not updated to reflect the new logic)? It looks quite bizarre. We are taking some count of addresses of one query as a partitioning key in another one to count the token_1 reserve?

I would simply order by created_at (which is a column that was added to 3959044.